streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -588,7 +598,7 @@ private void completeAllProcessableWork() {
     }
 
     private void commit(final Map<TopicPartition, OffsetAndMetadata> offsets) {
-        if (eosEnabled) {
+        if (processingMode == EXACTLY_ONCE_ALPHA || processingMode == EXACTLY_ONCE_BETA) {
```

Comment:
TBH, Since we are adding public flags, could we just make the `StreamThread` helper functions public so that we don't need to manually write check every time?

Reply:
Current `StreamThread` helper function are base on `StreamsConfig` and don't do anything with `ProcessingMode` -- and `streamsConfig` is only available in the constructor of TTD on purpose.

Reply:
@mjsax has the fair point here, I think we can defer for future PRs to have a `StreamsUtil` like function for such static functions.